### PR TITLE
Signup: Add privacy when we add domains

### PR DESCRIPTION
**Pull request by scruffian**
#### Changes proposed in this Pull Request

This removes the privacyItem from the plans step. Now all domains come with privacy we can just add it when the domain is added to the cart.
#### Testing instructions

* Check that the signup flows work as before
* Try the flow at http://calypso.localhost:3000/start/plan-no-domain and confirm that the plan is added the cart at checkout.

The other changes originally proposed are now here: https://github.com/Automattic/wp-calypso/pull/29663

### CODE REVIEW

**Comment by scruffian** [comment]
[Update the way signup dependencies are defined](https://trello.com/c/2wCy3CrA/334-update-the-way-signup-dependencies-are-defined)

**Comment by scruffian** [comment]
We can totally remove the privacyItem from signup now - see p99Zz8-uw-p2 

cc @aidvu

**Comment by aidvu** [comment]
Looking at the `step-actions`, we already auto-add privacy when there's a domain item, so no need for `privacyItem` from what I gather.

**Comment by scruffian** [comment]
> Looking at the step-actions, we already auto-add privacy when there's a domain item, so no need for privacyItem from what I gather.

@aidvu - do you think it makes more sense for this logic to live in the API?

**Comment by aidvu** [comment]
> @aidvu - do you think it makes more sense for this logic to live in the API?

I don't. It's already in `step-actions` and it works. Changing cart to auto-add this will add more trouble, because we have to do special checks whether this is something that should be auto-added, when it's super easy to just add it in the `step-actions`.

**Comment by scruffian** [comment]
I have removed `privacyItem` from the other plans steps.

**Comment by aidvu** [comment]
Would be good to check the e2e tests failing, main checkout flow.

File: client/signup/config/steps-pure.js [hunk 1]
```diff
@@ -147,15 +147,15 @@ export function generateSteps( {
 		plans: {
 			stepName: 'plans',
 			apiRequestFunction: addPlanToCart,
-			dependencies: [ 'siteSlug', 'siteId', 'domainItem' ],
```
Comment (aidvu) [inline client/signup/config/steps-pure.js]: Not sure if smart. Now you have to check all steps to not pass the `siteId`. I'd rather do this in a different PR.

**Comment by aidvu** [comment]
Please confirm why the e2e tests are failing. Seems the main flow isn't working after the change.

**Comment by scruffian** [comment]
I think the tests are failing because the server patch isn't merged.

File: client/signup/config/steps-pure.js (hunk 1 above)
Comment (scruffian) [reply client/signup/config/steps-pure.js]: Makes sense

Comment (scruffian) [reply client/signup/config/steps-pure.js]: Done in #29663
